modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableRowSkinBase.java

```diff
@@ -542,18 +542,18 @@ void updateCells(boolean resetChildren) {
 
         // update children of each row
         if (fixedCellSizeEnabled) {
-            // we leave the adding / removing up to the layoutChildren method mostly,
-            // but here we remove any children cells that refer to columns that are
-            // not visible
+            // we leave the adding / removing up to the layoutChildren method mostly, but here we remove any children
+            // cells that refer to columns that are removed or not visible.
             List<Node> toRemove = new ArrayList<>();
             for (Node cell : getChildren()) {
-                if (! (cell instanceof IndexedCell)) continue;
-                if (!getTableColumn((R)cell).isVisible()) {
+                if (!(cell instanceof IndexedCell)) continue;
+                TableColumnBase<T, ?> tableColumn = getTableColumn((R) cell);
+                if (!getVisibleLeafColumns().contains(tableColumn) || !tableColumn.isVisible()) {
```

Comment:
coming back, now that I understand the overall logic :)

checking column.isVisible is not necessary: not being contained in the visibleLeafColumns implies that it is either not visible or not in the column hierarchy of table's columns.

Reply:
oh, right. It's called getVisibleLeafColumns() for a reason. 😃 
I will remove the visibility check.